src/main/java/uk/gov/companieshouse/filinghistory/consumer/transformrules/functions/FormatDate.java

```diff
@@ -38,12 +41,22 @@ public FormatDate(ObjectMapper objectMapper) {
     @Override
     protected void doTransform(JsonNode source, TransformTarget target, List<String> arguments,
             Map<String, String> context) {
-        target.objectNode().put(target.fieldKey(), format(target.fieldValue()));
+        target.objectNode().put(target.fieldKey(), formatIfRequired(target.fieldValue()));
     }
 
     public String format(String nodeText) {
+        return doFormat(nodeText);
+    }
+
+    public String formatIfRequired(String nodeText) {
+        if (ISO_DATE_PATTERN.matcher(nodeText).matches()) {
```

Comment:
This blows up if passed a null string, we can just return nodeText if it is null or empty before creating the matcher.
```java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null```